Last fixes

edouardproust opened the pull request:
- Updated root Makefile and sub-Makefiles (including the one of libft) to force recompiling libft, MLX42 and mandatory / bonus if the root Makefile is updated
- Added a check for folder textures paths in parsing
- Deleted the textures with transparency in `bonus/assets/` and updated the maps accordingly
- Added `bonus/assets/door.png` and updated the maps accordingly
- Replaces `.supp` file in `test/` with the new one
- Added check of the library leaks in `test/` > `lib_leaks_proof/`
- Norminette (headers, removed `TODO` and `DEBUG` comments)

Commit:
fixes

Commit:
norminette mandatory

Commit:
norminette bonus

Commit:
fix color struct name

Skoteini-42 commented:
Looks fine except the issue with the trimming with the newline as we talked about.

I want to add  3 maps to the bonus : a large one, one with central rooms and a map simulating a maze with many doors. Also i will decrease the MM scsale to 10 from 14, as this would show a very large MM if the map is large, blocking 1/4 of the screen.

(no changes in the code or anything else that can affect the code)

So I am goign to wait for your fixes to be pushed, then add the 3 maps inside

Commit:
fix newline problem on parsing

edouardproust commented:
- I changed the type of `t_color->rgb` to an `int` instead of an `unsigned int`.
- Also added a check to trim the newline in parsing: i trim only if the last charatecter is `\n` (in `parsing/parse_metadata.c::parse_metadata()`)
- I created 3 .cub files in `mandatory/maps` to check: `no_grid.cub`, `no_color.cub` and `color_id_invalid.cub` but i recommend you to do your own tests / maps to be sure, thanks

Skoteini-42 committed:
added 3 bonus maps and changed MM scale to 10

Skoteini-42 commented:
Seems fixed, nice ! Accepted
